lib/pingurl.js

```diff
@@ -12,51 +12,54 @@ const TIMEOUT_MS = 5000;
 const TIMEOUT_WAIT_MS = 1000;
 const MAX_ATTEMPTS = 3;
 
+const sleep = ms => new Promise(r => setTimeout(r, ms));
+const getAgent = (url, a) => (a ? (url.match(/^https/) ? a.https : a.http) : undefined);
```

Comment:
Looks like you're getting the schema not the agent, is this a misnomer?

Reply:
oh I see, the option is called agent; fair enough!

Reply:
Yeah, it's an node `http.request` option.  (And apparently some lambda-housekeeping thing I stumbled on years ago, that you need to cleanup your agents with every lambda execution).